linkml/utils/deprecation.py

```diff
@@ -245,6 +245,19 @@ def warn(self, **kwargs):
         ),
         recommendation="Update to use `gen-doc`",
     ),
+    Deprecation(
+        name="gen-yuml",
+        # the last update to any code in yumlgen.py was in v1.8.7
+        # we can start considering it as deprecated from this version
+        deprecated_in=SemVer.from_str("1.8.7"),
```

Comment:
The way I got this for `gen-markdown` deprecation was by checking in what release we had last modified the generator module/py file.